include/ps/internal/message.h

```diff
@@ -90,6 +107,8 @@ struct Node {
   std::string hostname;
   /** \brief the port this node is binding */
   int port;
+  /** \brief whether this node is created by failover */
+  std::atomic<bool> is_recovery;
```

Comment:
is_recovery is read at Postoffice::is_recovery() (postoffice.h line 137), which may be from another thread.